test/SILGen/keypaths.swift

```diff
@@ -352,6 +399,45 @@ func subscripts<T: Hashable, U: Hashable>(x: T, y: U, s: String) {
 
   _ = \Subscripts<T>.[Bass()]
   _ = \Subscripts<T>.[Treble()]
+
+  _ = \SubscriptDefaults1.[]
+  _ = \SubscriptDefaults1.[0]
+  _ = \SubscriptDefaults1.[0, 0]
+  _ = \SubscriptDefaults1.[0, 0, 0]
+  _ = \SubscriptDefaults1.[false]
+  _ = \SubscriptDefaults1.[false, bool: false]
+  _ = \SubscriptDefaults1.[bool: false, 0]
+  _ = \SubscriptDefaults1.[bool: false, 0, 0]
+  
+  _ = \SubscriptDefaults2.[]
+  _ = \SubscriptDefaults2.[0]
+  _ = \SubscriptDefaults3.[]
+  _ = \SubscriptDefaults3.[0]
+  _ = \SubscriptDefaults4.[x: 0]
+  _ = \SubscriptDefaults4.[x: 0, y: 0]
+  _ = \SubscriptDefaults5.[x: ""]
```

Comment:
Could you make `\SubscriptDefaults5.[x: ""]` a FileCheck test? Just to make sure we correctly handle the interaction between generic params and caller-side defaults.